src/lib/viewers/doc/DocBaseViewer.js

```diff
@@ -1000,6 +1007,24 @@ class DocBaseViewer extends BaseViewer {
             // Add page IDs to each page after page structure is available
             this.setupPageIds();
         }
+
+        this.initThumbnails();
+    }
+
+    initThumbnails() {
+        this.thumbnailsSidebar.init({
+            totalItems: this.pdfViewer.pagesCount,
+            itemHeight: 115,
+            containerHeight: this.docEl.clientHeight,
+            marginTop: 7.5,
+            marginBottom: 7.5,
+            renderItemFn: (page) => {
+                const thumbnail = document.createElement('button');
```

Comment:
We're creating/trashing a lot of objects, but given the low number of living objects at any given time, it should be okay.